src/Widget.php

```diff
@@ -87,32 +87,19 @@ final public static function end(): string
      *
      * @return static The widget instance.
      */
-    final public static function widget(array|callable|string $config = []): static
+    final public static function widget(array $constructorArguments = [], array $config = []): static
     {
-        if (is_array($config) && !array_key_exists('class', $config)) {
-            $config['class'] = static::class;
+        if ($config === []) {
+            $config = [
+                '__construct()' => $constructorArguments,
+            ];
+        } elseif ($constructorArguments !== []) {
+            throw new InvalidArgumentException();
         }
 
-        return WidgetFactory::createWidget($config);
-    }
+        $config['class'] = static::class;
```

Comment:
It was previously possible to override class via `class` in config passed explicitly.

Reply:
Seems, it's bad hack. And now it throws error, because return type of `widget()` changed to `static`.